### PR TITLE
Task Drivers.

### PR DESCRIPTION
**Pull request by HubertHuckevoll**
Make TaskMax and B&F task driver compile. DOS5 task driver doesnt't compile and seems to be not fully implemented.

### CODE REVIEW

File: Installed/Driver/Task/NonTS/Makefile [hunk 1]
```diff
@@ -1,64 +1,34 @@
-##############################################################################
```
Comment (bluewaysw) [inline Installed/Driver/Task/NonTS/Makefile:1]: Is it necessary to replace this by a generated file?

File: bin/basebox.conf [hunk 2]
```diff
@@ -25,7 +25,7 @@ fullscreen=false
 fulldouble=false
 fullresolution=original
 windowresolution=original
-output=surface
+output=opengl
```
Comment (bluewaysw) [inline bin/basebox.conf:28]: This change is unrelated here, please revert this change for the PR.

File: bin/shiny [hunk 3]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
```
Comment (bluewaysw) [inline bin/shiny:1]: Unrelated addition, please remove from PR.

**Review by bluewaysw — CHANGES_REQUESTED** [review]
Please clean up, there are unrealted changes.

**Comment by javiergutierrezchamorro** [comment]
Any way to get its binaries for being able to test the PR?

**Comment by coldacid** [comment]
@HubertHuckevoll could you update your branch and make the cleanups requested by @bluewaysw so this could be merged? I'd love to get to test this out. Thanks!

File: Installed/Driver/Task/BNF/dependencies.mk [hunk 4]
```diff
@@ -0,0 +1,45 @@
+bnf.rdef: generic.uih product.uih \
+                /home/konstantinmeyer/github/pcgeos/Driver/Task/Common/task.ui
```
Comment (coldacid) [inline Installed/Driver/Task/BNF/dependencies.mk:2]: This should probably be relative, not an absolute path that would only make sense on one particular person's machine.

File: Installed/Driver/Task/TaskMax/dependencies.mk [hunk 5]
```diff
@@ -0,0 +1,46 @@
+taskmax.rdef: generic.uih product.uih \
+                /home/konstantinmeyer/github/pcgeos/Driver/Task/Common/task.ui
```
Comment (coldacid) [inline Installed/Driver/Task/TaskMax/dependencies.mk:2]: This also should be a relative path.

File: Installed/Driver/Task/NonTS/Makefile (hunk 1 above)
Comment (coldacid) [reply Installed/Driver/Task/NonTS/Makefile:1]: The original NonTS Makefile was also automatically generated from the original mkmf by the look of things. It's probably okay for it to have been regenerated with the new mkmf, especially as it is consistent with the other task driver makefiles.

Comment (coldacid) [reply Installed/Driver/Task/NonTS/Makefile:1]: There probably should be a Driver/Task/NonTS/local.mk file though, for the `PROTOCONST` and `LIBNAME` values which no longer exist in this file.

**Comment by coldacid** [comment]
The B&F driver should probably be added to Installed/Makefile as well, similar to the Taskmax driver.

**Comment by coldacid** [comment]
@HubertHuckevoll @bluewaysw I have an updated version of this branch at https://github.com/coldacid/pcgeos/tree/Tasks which tackles the requested cleanups, as well as includes the drivers and the task switch prefs panel in product build.

File: Installed/Driver/Task/BNF/dependencies.mk (hunk 4 above)
Comment (bluewaysw) [reply Installed/Driver/Task/BNF/dependencies.mk:2]: Yes, there is an issue in the generation tools when using mkmf/mkmf depend. For now manual correction after automatic generation of the files is the way to go.

**Comment by bluewaysw** [comment]
> @HubertHuckevoll @bluewaysw I have an updated version of this branch at https://github.com/coldacid/pcgeos/tree/Tasks which tackles the requested cleanups, as well as includes the drivers and the task switch prefs panel in product build.

Hey @coldacid thx a lot for the effort and support, looks very good. Did you had a chance to test the task switcher drivers in real environment so they are at least basically working? Would you mind to open an new PR from you changes?

**Comment by coldacid** [comment]
@bluewaysw I've not tested yet because I have a bit of trouble getting things into my 86Box environment when FAT is involved, but I'll give it a shot. I'll only be able to test the Taskmax driver in DR-DOS 7 mode, though.

**Comment by coldacid** [comment]
@bluewaysw looks like the TaskMAX / Novell DOS Task Manager driver fails softly. I don't know if it's something with my Novell DOS 7 config inside 86Box, but during DosExecSuspend (called by the common task driver code) something fails when telling drivers to suspend themselves. No clue which one, or why, since es:di is just an empty string on return from DosExecSuspend, so the error popup shown by the driver on failure doesn't tell me anything more than "Unable to suspend" GEOS.

At least we know it doesn't crash on failure (although the screen fails to be entirely repainted on unsuspend) so it's probably safe to merge, with the caveat that GEOS itself might not be able to suspend itself for task switching.

**Comment by coldacid** [comment]
@bluewaysw @HubertHuckevoll new PR at #372
